### DISCUSSION TITLE
Binding IImage to memory instead of file?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [AvaloniaUI/Avalonia#16943] [Q&A]
Can I bind a IImage directly to the ViewModel's WriteableBitmap Property?

**Comment by stevemonaco** [comment]
Yes, `WriteableBitmap` implements `IImage` in its hierarchy.

Note if you're binding this to an `<Image x:Name="myImage" Source="{Binding Wb}" />`, then you need to manually invalidate the `Image` control (using `myImage.InvalidateVisual();`) when you update the `WriteableBitmap`. You'll need to do this from code-behind, which can be a bit awkward. Messaging or assigning an `Action` callback into the VM from code-behind are two ways to solve.
